tensorflow_probability/python/positive_semidefinite_kernels/polynomial.py

```diff
@@ -0,0 +1,160 @@
+        known as the degree) of the polynomial function. A value of `None` is
+        treated like 1 which results in a linear kernel.
+        Default Value: `None`
+      feature_ndims: Python `int` number of rightmost dims to include in kernel
+        computation.
+        Default Value: 1
+      validate_args: If `True`, parameters are checked for validity despite
+        possibly degrading runtime performance
+        Default Value: `False`
+      name: Python `str` name prefixed to Ops created by this class.
+        Default Value: `'Polynomial'`
+    """
+    with tf.name_scope(name, values=[bias_variance, slope_variance, exponent]) as name:
+      dtype = dtype_util.common_dtype([bias_variance, slope_variance, exponent], tf.float32)
+      if bias_variance is not None:
+        bias_variance = tf.convert_to_tensor(
+          bias_variance, name='bias_variance', dtype=dtype)
+      self._bias_variance = _validate_arg_if_not_none(
+        bias_variance, tf.assert_positive, validate_args)
+      if slope_variance is not None:
+        slope_variance = tf.convert_to_tensor(
+          slope_variance, name='slope_variance', dtype=dtype)
+      self._slope_variance = _validate_arg_if_not_none(
+        slope_variance, tf.assert_positive, validate_args)
+      if exponent is not None:
+        exponent = tf.convert_to_tensor(
+          exponent, name='exponent', dtype=dtype)
+      self._exponent = _validate_arg_if_not_none(
+        exponent, tf.assert_positive, validate_args)
+      tf.assert_same_float_dtype(
+        [self._bias_variance, self._slope_variance, self._exponent])
+    super(Polynomial, self).__init__(
+      feature_ndims, dtype=dtype, name=name)
+
+  @property
+  def bias_variance(self):
+    """Variance on bias parameter."""
+    return self._bias_variance
+
+  @property
+  def slope_variance(self):
+    """Variance on slope parameter."""
+    return self._slope_variance
+
+  @property
+  def exponent(self):
+    """Exponent of the polynomial term."""
+    return self._exponent
+
+  def _batch_shape(self):
+    scalar_shape = tf.TensorShape([])
+    return tf.broadcast_static_shape(
+      tf.broadcast_static_shape(
+        scalar_shape if self.slope_variance is None else self.slope_variance.shape,
+        scalar_shape if self.bias_variance is None else self.bias_variance.shape),
+      scalar_shape if self.exponent is None else self.exponent.shape)
+
+  def _batch_shape_tensor(self):
+    return tf.broadcast_dynamic_shape(
+      tf.broadcast_dynamic_shape(
+        [] if self.slope_variance is None else tf.shape(self.slope_variance),
+        [] if self.bias_variance is None else tf.shape(self.bias_variance)),
+      [] if self.exponent is None else tf.shape(self.exponent))
+
+  def _apply(self, x1, x2, param_expansion_ndims=0):
+    dot_prod = util.sum_rightmost_ndims_preserving_shape(
+      tf.multiply(x1, x2), ndims=self.feature_ndims)
```

Comment:
we generally prefer to use the overloaded operators where available (i.e. x1 * x2)